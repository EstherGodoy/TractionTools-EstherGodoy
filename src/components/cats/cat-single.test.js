import React from 'react';
import ReactDOM from 'react-dom';
import CatSingle from './cat-single';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure, render } from 'enzyme';
configure({ adapter: new Adapter() });


it('component renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CatSingle />, div);
});

test("single cat snapshot test", (() => {
  const component = renderer.create(
    <CatSingle/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}))

it('loads required cat data (image)', () => {
  const wrapper = mount(<CatSingle/>);
  //has figure
  expect(wrapper.find('figure').length).toEqual(1);
  //background image loaded
  expect(wrapper.find('figure').get(0).props.style.backgroundImage.length > 0);
});
