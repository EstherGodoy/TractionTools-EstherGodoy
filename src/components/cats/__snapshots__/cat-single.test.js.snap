// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single cat snapshot test 1`] = `
<li
  className=""
>
  <figure
    style={
      Object {
        "backgroundImage": "url(false)",
      }
    }
  >
    <button
      className="add"
      onClick={[Function]}
    >
      Add To Family
    </button>
  </figure>
</li>
`;
